server/routes/deck.js

```diff
@@ -0,0 +1,73 @@
+const router = require('express').Router();
+const mongoose = require('mongoose');
+require('../models/DeckSchema');
+const Deck = mongoose.model('Deck');
+
+router.post('/createDeck', function(req, res) {
+    const deck = new Deck();
+    deck.title = req.body.title;
+    
+    let shortId;
+    do {
+        shortId = Math.random().toString(36).substr(2, 6).toUpperCase()
+    } while(Deck.findOne({'shortId': shortId}).length);
+    deck.shortId = shortId;
+
+
+    // narazie tak to zostawie, a potem pewnie będzie brane z uwierzytelniania
+    deck.userId = req.body.userId;
+
+    deck.save().then(function(deck) {
+        res.status(201);
+        res.json({'id': deck.shortId});
+    }).catch(function(error) {
+        res.send(error.message);
+    });
+});
+
+router.get('/getDeck/:id', function(req, res) {
+    const deck = Deck.findOne({'shortId': req.params.id});
+    deck.then(function (deck) {
```

Comment:
Możesz to robić jeden po drugim tzw. chained promises, więc .then możesz zrobić od razu po wywołaniu funkcji